src/Cake.Prca/Issues/CodeAnalysisIssue.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace Cake.Prca.Issues
+{
+    using System;
+    using Core.IO;
+
+    /// <summary>
+    /// Base class for a code analysis issue.
+    /// </summary>
+    public class CodeAnalysisIssue : ICodeAnalysisIssue
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CodeAnalysisIssue"/> class.
+        /// </summary>
+        /// <param name="filePath">The path to the file affacted by the issue.
+        /// The path needs to be relative to the repository root.</param>
+        /// <param name="line">The line in the file where the issues has occurred.</param>
+        /// <param name="message">The message of the code analysis issue.</param>
+        /// <param name="priority">The priority of the message used to filter out issues if there are more issues than
+        /// should be posted.</param>
+        /// <param name="rule">The rule of the code analysis issue.</param>
+        public CodeAnalysisIssue(string filePath, int line, string message, int priority, string rule)
+        {
+            filePath.NotNullOrWhiteSpace(nameof(filePath));
+            line.NotNegativeOrZero(nameof(line));
```

Comment:
@stffabi Which leads to the question of what the scope is. Is it only issues which can be mapped to a line (current implementation) or all issues. You preference is to let the pull request system handle this?